### ISSUE TITLE
Why apptainer creates two mount namespaces when launching a container?

### ISSUE THREAD

**Issue by cyyzero** [apptainer/apptainer#1058]
### Version of Apptainer

What version of Apptainer (or Singularity) are you using? Run

```sh
$ apptainer --version
apptainer version 1.0.0+7-gab8980eb8
```

### question

When I launch a container via apptainer, I found that `starter-suid` will create two mount namespace

```sh
# launch a container 
$ apptainer shell xxx.sif

# list all namespaces
$ sudo lsns
        NS TYPE  NPROCS   PID USER   COMMAND
4026531836 pid      167     1 root   /usr/lib/systemd/systemd --switched-root --system --deserialize 22
4026531837 user     167     1 root   /usr/lib/systemd/systemd --switched-root --system --deserialize 22
4026531838 uts      167     1 root   /usr/lib/systemd/systemd --switched-root --system --deserialize 22
4026531839 ipc      167     1 root   /usr/lib/systemd/systemd --switched-root --system --deserialize 22
4026531840 mnt      163     1 root   /usr/lib/systemd/systemd --switched-root --system --deserialize 22
4026531860 mnt        1    49 root   kdevtmpfs
4026531992 net      167     1 root   /usr/lib/systemd/systemd --switched-root --system --deserialize 22
4026532356 mnt        1   700 chrony /usr/sbin/chronyd
4026532357 mnt        1 30374 cyy    Apptainer runtime parent
4026532358 mnt        1 30393 cyy    /bin/bash --norc
```

After I cursory reading of the code of  `starter-suid`, I found that apptainer creates tow nested mount namespaces when initializing the container environment. What is the reason for this? Because I noticed that container engines like docker/podman, which rely on runc, only create one layer of namespace to isolate containers.

### How did you install Apptainer

```sh
$ ./mconfig
$ cd builder; make
```

**Comment by GodloveD** [comment]
It is my understanding that the starter program needs to set up a mount namespace and then creates the mount namespace for the unprivileged user.  I believe this is necessary to drop privs properly with the suid bit workflow.  Not sure how relevant it is anymore with an unprivileged installation.  @DrDaveD or @cclerget can jump in with more details if needed.

**Comment by DrDaveD** [comment]
Yes it is necessary for good security in the suid mode.  It's not needed in non-suid but it would be very difficult to use a different base architecture for the two different modes.

On the other hand, in any case there would need to be 2 user namespaces.  The first one maps the original user to root (which is needed for doing mounts and other normally privileged operations), and the second one maps the user id back from root to the original user.  This is different from what docker/podman usually do.
